draft-tjhai-ipsecme-hybrid-qske-ikev2-04.xml

```diff
@@ -0,0 +1,994 @@
+			many of these algorithms are relatively new and there may be a need to
+			hedge the security risk with multiple key exchange data from several
+			distinct QSC algorithms.
+            </t>
+
+            <t>
+			The secrets established from each key exchange are combined in a way
+			such that should the post-quantum secrets not be present, the derived
+			shared secret is equivalent to that of the standard IKEv2; on the
+			other hand, a post-quantum shared secret is obtained if both classical
+			and post-quantum key exchange data are present.  This framework also
+			applies to key exchanges in IKE Security Associations (SAs) for
+			Encapsulating Security Payload (ESP) <xref target="RFC4303"/> or
+			Authentication Header (AH) <xref target="RFC4302"/>, i.e. Child SAs,
+			in order to provide a stronger guarantee of forward security.</t>
+
+            <t>
+			Some post-quantum key exchange payloads may have size larger than
+			the standard maximum transmission unit (MTU) size, and therefore there could be issues with
+			fragmentation at IP layer.  IKE does allow transmission over TCP
+			where fragmentation is not an issue <xref target="RFC8229"/>;
+			however, we believe that a UDP-based solution will be required
+			too.  IKE does have a mechanism to handle fragmentation within
+			UDP <xref target="RFC7383"/>, however that is only applicable to
+			messages exchanged after the IKE_SA_INIT.  To use this mechanism,
+			we use the IKE_INTERMEDIATE exchange as outlined in
+			<xref target="I-D.ietf-ipsecme-ikev2-intermediate"/>.  With this
+			mechanism, we do an initial key exchange, using a smaller, possibly
+			non-quantum resistant primitive, such as ECDH.  Then, before we do
+			the IKE_AUTH exchange, we perform one or more IKE_INTERMEDIATE exchanges,
+			each of which includes a secondary key exchange.  As the IKE_INTERMEDIATE
+			exchange is encrypted, the IKE fragmentation protocol RFC7383
+			can be used. The IKE SK_* values are updated after each exchange,
+			and so the final IKE SK_* values depend on all the key exchanges,
+			hence they are secure if any of the key exchanges are secure.</t>
+
+            <t>
+			Note that readers should consider the approach in this document as
+			providing a long term solution in upgrading the IKEv2 protocol to
+			support post-quantum algorithms.  A short term solution to make IKEv2
+			key exchange quantum secure is to use post-quantum pre-shared keys as
+			discussed in <xref target="I-D.ietf-ipsecme-qr-ikev2"/>.</t>
+
+            <t> Note also, that the proposed approach of performing multiple successive key exchanges
+            in such a way that resulting session keys depend on all of them is not limited
+            to achieving quantum resistance only. In can also be used when all 
+            the performed key exchanges are classical (EC)DH ones, but for some reasons
+            (e.g. policy requirements) it is essential to perform multiple of them.
```

Comment:
How about replacing the last sentence with the following:

If for some reasons (e.g. policy requirements) it is essential to perform more than one classical (EC)DH key exchanges, the proposed approach can also serve this use-case.

Reply:
No objection. Just want to emphasize that in an extreme case all key exchanges can be classical (the other extreme case when all of them, including that in IKE_SA_INIT are PQ). How about:

If for some reasons (e.g. policy requirements) it is essential to perform more than one classical (EC)DH key exchanges, the proposed approach can also serve this use-case, including situation when all the performed key exchanges are classical ones.

Reply:
Yep, happy with this.

Reply:
Do you want me to change the file ot you'll do it?